### PR TITLE
domins-recipe_app setting

### PR DESCRIPTION
**Pull request by Bleu-Yves-Sopoude**
In this project, we(@geekelo and @Bleu-Yves-Sopoude ) implemented all these requirements below:

    - You should follow the layout of the wireframes provided. You should personalize the rest of the design including colors, typographies, spacings, etc.
    - Login page and registration page:
        Should be built with Devise.
    - Food list (only for a group that has less than 3 members):
        - Should display a list of food added by the logged-in user as in the wireframe (display also quantity of a given food).
        - Should lead to a form that allows users to add new food.
    - Recipes list:
        - Should display a list of recipes created by the logged-in user as in the wireframe.
        - Should lead to recipe details.
        If the user is the owner of the recipe, should allow the user to delete it.
    - Public recipe list:
        - Should display a list of all public recipes ordered by newest as in the wireframe.
        - Should lead to recipe details.
    - Recipe details:
        - Should display a toggle button that allows for a recipe to be made public or private.
        - If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
        - If the user is the owner of the recipe, should lead to the form that allows the user to add new food.
   - General shopping list view (only for a group that has less than 3 members):
        - Should show the list of food that is missing for all recipes of the logged-in user (compare the list of food for all recipes with the general food list of that user).
        - Should count the total food items and total price of the missing food.
    - Make sure there are no N+1 queries happening.
    - Create a navigation menu that allows users to open all of the pages you created.
    - Write unit and integration tests

### CODE REVIEW

**Comment by rloterh** [comment]
Hi @Bleu-Yves-Sopoude @geekelo 

Kindly ensure to add a professional ReadMe ASAP!.(**Required**)

It should follow [this template](https://github.com/microverseinc/readme-template)

Cheers!

File: app/controllers/general_shopping_lists_controller.rb [hunk 1]
```diff
@@ -0,0 +1,8 @@
+def index
+  @recipe_foods = RecipeFood.joins(:food)
+    .select('foods.name,
+                                     SUM(recipe_foods.quantity) as total_quantity,
+                                     SUM(recipe_foods.value) as total_value')
+    .group('foods.name')
+    .includes(:food)
+end
```
Comment (rloterh) [inline app/controllers/general_shopping_lists_controller.rb:8]: - [ ] An error is thrown in an attempt to generate a shopping list, like below. Let's ensure to fix this:

image

File: config/routes.rb [hunk 2]
```diff
@@ -1,10 +1,37 @@
...
+  get 'general_shopping_lists/index'
+  get 'recipe_foods/index'
+  get 'recipe_foods/show'
+  get 'recipe_foods/new'
+  get 'recipe_foods/create'
+  get 'recipe_foods/edit'
+  get 'recipe_foods/update'
+  get 'recipe_foods/destroy'
+  get 'public_recipes/index'
+  devise_for :users
```
Comment (rloterh) [inline config/routes.rb:11]: - [ ] An error is thrown in an attempt to logout. Let's ensure to fix it:

image

File: README.md [hunk 3]
```diff
@@ -1,24 +1,138 @@
...
+```sh
+  git clone https://github.com/Bleu-Yves-Sopoude/socialize.git
+```
+
+### Usage
+
+To run the project, execute the following command:
+
+```sh
+  ruby socialize.rb
```
Comment (rloterh) [inline README.md:75]: - [ ] Descriptions under the _Getting Started_ section does not match this project. Let's ensure to provide appropriate _Getting Started_ description/instructions on how to _set up_, _run_ and _test_ the app, for instance using the command `rails server` and `rspec` to run and test the app respectively.

**Review by rloterh — CHANGES_REQUESTED** [review]
Hi  Geekelo & MrBlue, 

You did extremely great on this project! 👏 👏 
There are however a few issues that you still need to work on to go to the next project but you are almost there! 
🚦 __🏃 🏃‍♀️  🏡  🏚️ 🌴

## Highlights 👍  
- Authentication with Devise gem properly implemented!
- App schema closely follows given ERD!
- Properly addressed N+1 problem
- All users can view public recipes!
- ReadMe looks awesome!

## Required Changes ♻️
_Kindly check the inline comments under the review._

Cheers and Happy coding!👏 💻 🍷

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by mwanawabangona — APPROVED** [review]
Hi @geekelo & @Bleu-Yves-Sopoude 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:
Congratulations!  


## Highlights
*   No linter errors :heavy_check_mark:
*   Repo looks professional :briefcase:
*   Solid pull request scroll
*   Good quality commits
*   Good job Implementing the requested changes !! :clap: :clap: :clap:

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**